### PR TITLE
chore: bump dependencies

### PR DESCRIPTION
**Pull request by smira**
dependabot + go-mod-outdated

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5700)

### CODE REVIEW

File: api/resource/resource.proto [hunk 1]
```diff
@@ -33,6 +33,7 @@ message Metadata {
   google.protobuf.Timestamp created = 8;
   google.protobuf.Timestamp updated = 9;
   repeated string finalizers = 6;
+  map<string, string> labels = 10;
```
Comment (DmitriyMV) [inline api/resource/resource.proto:36]: Huh? Should't it be in labels PR?

Comment (smira) [reply api/resource/resource.proto:36]: Talos has its own gRPC API for COSI for historical reasons. I'm not really introducing labels into Talos here, but this change is required to make the code compile (as Metadata proto structure should implement COSI interface).

Comment (frezbo) [reply api/resource/resource.proto:36]: so this change is from the recently added label selectors in COSI, right?

**Review by frezbo — APPROVED** [review]

File: api/resource/resource.proto (hunk 1 above)
Comment (smira) [reply api/resource/resource.proto:36]: yep, as COSI got bumped, it has now labels in COSI gRPC API (vs. Talos gRPC resources API), so it requires Talos Metadata proto to have these labels

**Review by Unix4ever — APPROVED** [review]

**Comment by Unix4ever** [comment]
Looks like there are some places which rely on `Spec` being a value, not a pointer.

**Comment by smira** [comment]
> Looks like `LinkSpec` controller is broken as it expects the spec to be value not a pointer.

thanks, yep, fixed now

**Comment by smira** [comment]
fixed one more problem with grpc-proxy and new gRPC library

**Comment by smira** [comment]
and one more issue with `Spec()` being a pointer now, but the code gets better

**Pull request merged**